File: backend/src/main/java/com/todolist/app/models/Todo.java
package com.todolist.app.models;

import com.todolist.app.dtos.Status;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "todos")
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String title;

    private String content;

    private LocalDateTime startDate;

    private LocalDateTime endDate;

    @Enumerated(EnumType.STRING)
    private Status status;
}

File: frontend/src/types/todo.ts
export type TodoStatus = 'ONGOING' | 'LATE' | 'COMPLETED';

export interface TodoRequest {
    title: string;
    content: string;
    startDate: string;
    endDate: string;
}

export interface TodoResponse {
    id: number;
    title: string;
    content: string;
    startDate: string;
    endDate: string;
    status: TodoStatus;
}

// Copy the DTOs type to FE